cg/apps/sequencing_metrics_parser/parsers/bcl_convert.py

```diff
@@ -0,0 +1,93 @@
+"""This module parses metrics for files generated by the BCLConvert tool using the Dragen hardware."""
+import csv
+import xml.etree.ElementTree as ET
+import logging
+from pathlib import Path
+from typing import List, Union, Dict, Callable
+from cg.io.controller import ReadFile
+from cg.constants.demultiplexing import SampleSheetHeaderColumnNames
+from cg.constants.constants import FileFormat
+from cg.apps.sequencing_metrics_parser.models.bcl_convert import (
+    BclConvertAdapterMetrics,
+    BclConvertDemuxMetrics,
+    BclConvertQualityMetrics,
+    BclConvertRunInfo,
+    BclConvertSampleSheetData,
+)
+
+
+LOG = logging.getLogger(__name__)
+
+
+class BclConvertMetricsParser:
+    def __init__(
+        self,
+        bcl_convert_quality_metrics_path: Path,
+        bcl_convert_demux_metrics_file_path: Path,
+        bcl_convert_adapter_metrics_file_path: Path,
+        bcl_convert_sample_sheet_file_path: Path,
+        bcl_convert_run_info_file_path: Path,
+    ) -> None:
+        """Initialize the class."""
+        self.quality_metrics_path: Path = bcl_convert_quality_metrics_path
```

Comment:
```suggestion
        self.quality_metrics_path: Path = bcl_convert_quality_metrics_file_path
```
like the ones below?